pydantic/_internal/_config.py

```diff
@@ -0,0 +1,244 @@
+    loc_by_alias: bool
+    alias_generator: Callable[[str], str] | None
+    ignored_types: tuple[type, ...]
+    allow_inf_nan: bool
+
+    # new in V2
+    strict: bool
+    # whether instances of models and dataclasses (including subclass instances) should re-validate, default 'never'
+    revalidate_instances: Literal['always', 'never', 'subclass-instances']
+    ser_json_timedelta: Literal['iso8601', 'float']
+    ser_json_bytes: Literal['utf8', 'base64']
+    # whether to validate default values during validation, default False
+    validate_default: bool
+
+    def __init__(self, config: ConfigDict | dict[str, Any] | type[Any] | None, *, check: bool = True):
+        if check:
+            self.config_dict = prepare_config(config)
+        else:
+            self.config_dict = cast(ConfigDict, config)
+
+    @classmethod
+    def for_model(cls, bases: tuple[type[Any], ...], namespace: dict[str, Any], kwargs: dict[str, Any]) -> Self:
+        """
+        Build a new `ConfigDict` instance for a `BaseModel` based on (from lowest to highest)
+        - options defined in base
+        - options defined in namespace
+        - options defined via kwargs
+
+        :param bases: tuple of base classes
+        :param namespace: namespace of the class being created
+        :param kwargs: kwargs passed to the class being created
+        """
+
+        config_new = ConfigDict()
+        for base in bases:
+            config = getattr(base, 'model_config', None)
+            if config:
+                config_new.update(config.copy())
+
+        config_class_from_namespace = namespace.get('Config')
+        config_dict_from_namespace = namespace.get('model_config')
+
+        if config_class_from_namespace and config_dict_from_namespace:
+            raise PydanticUserError('"Config" and "model_config" cannot be used together', code='config-both')
+
+        config_from_namespace = config_dict_from_namespace or prepare_config(config_class_from_namespace)
+
+        if config_from_namespace is not None:
+            config_new.update(config_from_namespace)
+
+        for k in list(kwargs.keys()):
+            if k in config_keys:
+                config_new[k] = kwargs.pop(k)
+
+        return cls(config_new)
+
+    # we don't show `__getattr__` to type checkers so missing attributes cause errors
+    if not TYPE_CHECKING:
+
+        def __getattr__(self, name: str) -> Any:
+            try:
+                return self.config_dict[name]
+            except KeyError:
+                try:
+                    return config_defaults[name]
+                except KeyError:
+                    raise AttributeError(f'Config has no attribute {name!r}') from None
+
+    def core_config(self, obj: Any = None) -> core_schema.CoreConfig:
+        """
+        Create a pydantic-core config, `obj` is just used to populate `title` if not set in config.
+
+        We don't use getattr here since we don't want to populate with defaults.
+        """
```

Comment:
Is there a better way to handle `obj`? Could this function accept a fallback instead?

Reply:
I think this is fine for now.